src/Sarif.Viewer.VisualStudio/ErrorList/ErrorListService.cs

```diff
@@ -34,139 +37,168 @@ public class ErrorListService
 
         public static void ProcessLogFile(string filePath, string toolFormat, bool promptOnLogConversions, bool cleanErrors)
         {
-            // For now this is being done on the UI thread
-            // and is only required due to the message box being shown below.
-            // This will be addressed when https://github.com/microsoft/sarif-visualstudio-extension/issues/160
-            // is fixed.
-            ThreadHelper.ThrowIfNotOnUIThread();
+            var taskStatusCenterService = (IVsTaskStatusCenterService)Package.GetGlobalService(typeof(SVsTaskStatusCenterService));
+            var taskProgressData = new TaskProgressData
+            {
+                CanBeCanceled = false,
+                ProgressText = null,
+            };
 
-            SarifLog log = null;
+            string fileName = Path.GetFileName(filePath);
+
+            var taskHandlerOptions = new TaskHandlerOptions
+            {
+                ActionsAfterCompletion = CompletionActions.None,
+                TaskSuccessMessage = string.Format(CultureInfo.CurrentCulture, Resources.CompletedProcessingLogFileFormat, fileName),
+                Title = string.Format(CultureInfo.CurrentCulture, Resources.ProcessingLogFileFormat, fileName),
+            };
+
+            ITaskHandler taskHandler = taskStatusCenterService.PreRegister(taskHandlerOptions, taskProgressData);
 
-            string logText;
+            taskHandler.RegisterTask(ProcessLogFileAsync(filePath, toolFormat, promptOnLogConversions, cleanErrors));
+        }
+
+        public static async System.Threading.Tasks.Task ProcessLogFileAsync(string filePath, string toolFormat, bool promptOnLogConversions, bool cleanErrors)
+        {
+            SarifLog log = null;
+            string logText = null;
             string outputPath = null;
             bool saveOutputFile = true;
 
             if (toolFormat.MatchesToolFormat(ToolFormat.None))
             {
-                logText = File.ReadAllText(filePath);
-
-                Match match = MatchVersionProperty(logText);
-                if (match.Success)
+                using (FileStream fileStream = File.OpenRead(filePath))
+                using (StreamReader logStreamReader = new StreamReader(fileStream))
```

Comment:
Spot the bug:

        
        public StreamReader(Stream stream) 
            : this(stream, true) {
        }

        public StreamReader(Stream stream, bool detectEncodingFromByteOrderMarks) 
            : this(stream, Encoding.UTF8, detectEncodingFromByteOrderMarks, DefaultBufferSize, false) {
        }

Reply:
Well, the spec says the JSON must be UTF-8, so that's fine. Are you concerned about byte order marks?

---
In reply to: [494560156](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494560156) [](ancestors = 494560156)

Reply:
Nope: the StreamReader in the last constructor controls whether or not it closes the first argument.


---
In reply to: [494972041](https://github.com/microsoft/sarif-visualstudio-extension/pull/202#discussion_r494972041) [](ancestors = 494972041,494560156)